src/Agent.Plugins/PipelineCache/TarUtils.cs

```diff
@@ -195,14 +194,35 @@ public static class TarUtils
             }
         }
 
-        private static void SetProcessStartInfo(Process process, string processFileName, string processArguments)
+        private static void SetProcessStartInfo(ProcessStartInfo processStartInfo, string processFileName, string processArguments, string processWorkingDirectory)
         {
-            process.StartInfo.FileName = processFileName;
-            process.StartInfo.Arguments = processArguments;
-            process.StartInfo.UseShellExecute = false;
-            process.StartInfo.RedirectStandardInput = true;
-            process.StartInfo.RedirectStandardOutput = true;
-            process.StartInfo.RedirectStandardError = true;
+            processStartInfo.FileName = processFileName;
+            processStartInfo.Arguments = processArguments;
+            processStartInfo.UseShellExecute = false;
+            processStartInfo.RedirectStandardInput = true;
+            processStartInfo.RedirectStandardOutput = true;
+            processStartInfo.RedirectStandardError = true;
+            processStartInfo.WorkingDirectory = processWorkingDirectory;          
+        }
+
+        private static ProcessStartInfo GetCreateTarProcessInfo(string archiveFileName, string inputPath)
+        {
+            var processFileName = "tar";
+            inputPath = (inputPath.EndsWith(Path.DirectorySeparatorChar)) ? inputPath.Substring(0, inputPath.Length - 1) : inputPath;
+            var processArguments = $"-cf \"{archiveFileName}\" -C \"{inputPath}\" ."; // If given the absolute path, the GNU tar fails. The workaround is to start the tarring process in the temp directory, and simply speficy 'archive.tar'
```

Comment:
>var processArguments = $"-cf \"{archiveFileName}\" -C \"{inputPath}\" ."; // If given the absolute path, the GNU tar fails. The workaround is to start the tarring process in the temp directory, and simply speficy 'archive.tar' [](start = 12, length = 226)

Can't archiveFileName and inputPath be on different drives on Windows?

Reply:
They surely can.

Reply:
So then the comment doesn't make sense as we'll still have to use an absolute path for at least one of them, no?

Reply:
Good point. Let me articulate it nicely.